src/Testcontainers/Images/DockerfileArchive.cs

```diff
@@ -64,6 +66,54 @@ public DockerfileArchive(DirectoryInfo dockerfileDirectory, FileInfo dockerfile,
       _logger = logger;
     }
 
+    /// <summary>
+    /// Gets a collection of base images.
+    /// </summary>
+    /// <remarks>
+    /// This method reads the Dockerfile and collects a list of base images. It
+    /// excludes stages that do not correspond to base images. For example, it will not include
+    /// the second line from the following Dockerfile configuration:
+    /// <code>
+    ///   FROM mcr.microsoft.com/dotnet/sdk:6.0 AS build
+    ///   FROM build
+    /// </code>
+    /// </remarks>
+    /// <returns>An <see cref="IEnumerable{T}" /> of <see cref="IImage" />.</returns>
+    public IEnumerable<IImage> GetBaseImages()
+    {
+      const string imageGroup = "image";
+
+      var lines = File.ReadAllLines(Path.Combine(_dockerfileDirectory.FullName, _dockerfile.ToString()))
+        .Select(line => line.Trim())
+        .Where(line => !string.IsNullOrEmpty(line))
+        .Where(line => !line.StartsWith("#", StringComparison.Ordinal))
+        .Select(line => FromLinePattern.Match(line))
+        .Where(match => match.Success)
+        // Until now, we are unable to resolve variables within Dockerfiles. Ignore base
+        // images that utilize variables. Expect them to exist on the host.
+        .Where(match => !match.Groups[imageGroup].Value.Contains('$'))
+        .Where(match => !match.Groups[imageGroup].Value.Any(char.IsUpper))
+        .ToArray();
+
+      var stages = lines
+        .Select(line => line.Value)
+        .Select(line => line.Split(new [] { " AS " }, StringSplitOptions.RemoveEmptyEntries))
```

Comment:
Are lower cases of  `" AS "`somehow considered too?

Reply:
```suggestion
        .Select(line => line.Split(new [] { "AS", "As", "aS", "as" }, StringSplitOptions.RemoveEmptyEntries))
```